### ISSUE TITLE
[C10D] Standardize torch distributed launcher interface

### ISSUE THREAD

**Issue by wconstab** [pytorch/pytorch#124101]
Quoting @zdevito from the thread on #123992 
"I think it makes sense to eventually have a specification for what a particular launcher is required to provide to use torch distributed written in one place in the docs. It would say something like the following environment variables need to be set (RANK, SIZE, LOCAL_RANK, LOCAL_SIZE, TCP_STORE_ADDR, etc.). Then when we initialize torch distributed if a launcher launched without matching spec we could warn today and error tomorrow. That way we would start to encourage all launchers to become consistent in how they inform torch distributed about what is happening."

Basically, 
1) find a canonical place to document this info
2) add a warning if some of it is missing at runtime
